js/ai/src/prompt.ts

```diff
@@ -222,3 +561,158 @@ export function isExecutablePrompt(obj: any): boolean {
     !!(obj as ExecutablePrompt)?.stream
   );
 }
+
+export async function loadPromptFolder(
+  registry: Registry,
+  dir: string = './prompts',
+  ns: string
+): Promise<void> {
+  const promptsPath = resolve(dir);
+  if (existsSync(promptsPath)) {
+    const dirEnts = readdirSync(promptsPath, {
+      withFileTypes: true,
+      recursive: true,
+    });
+    for (const dirEnt of dirEnts) {
+      if (dirEnt.isFile() && dirEnt.name.endsWith('.prompt')) {
+        if (dirEnt.name.startsWith('_')) {
+          const partialName = dirEnt.name.substring(1, dirEnt.name.length - 7);
+          definePartial(
+            registry,
+            partialName,
+            readFileSync(join(dirEnt.path, dirEnt.name), {
+              encoding: 'utf8',
+            })
+          );
+          logger.debug(
+            `Registered Dotprompt partial "${partialName}" from "${join(dirEnt.path, dirEnt.name)}"`
+          );
+        } else {
+          // If this prompt is in a subdirectory, we need to include that
+          // in the namespace to prevent naming conflicts.
+          let prefix = '';
+          if (promptsPath !== dirEnt.path) {
+            prefix = dirEnt.path
+              .replace(`${promptsPath}/`, '')
+              .replace(/\//g, '-');
+          }
```

Comment:
Yeah as mentioned elsewhere, would be great if prompts could have slashes in their names before 1.0

Reply:
Noted.